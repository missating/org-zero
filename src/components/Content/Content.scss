.content-container {
  width: 100%;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  &__content {
    width: 80%;
    margin: 0 auto;

    &__heading {
      font-size: calc(5px + 2vmin);
      margin-top: 1.5rem;
      margin-bottom: 1rem;
    }

    &__items {
      display: flex;
      overflow: auto;
    }
  }
}

@media only screen and (max-width: 600px) {
  .content-container__content__items {
    display: block;
  }
}