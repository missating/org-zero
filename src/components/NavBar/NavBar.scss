.navbar-container {
  width: 100%;
  border-bottom: 1px solid #cccccc;
  cursor: pointer;

  &__content {
    width: 80%;
    margin: 0 auto;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0.5rem 0;

    &__items {
      display: flex;
      padding: 0;
      margin: 0;
      color: #4a90e2;

      li {
        list-style: none;
        margin-left: 1rem;
      }
    }
  }
}
