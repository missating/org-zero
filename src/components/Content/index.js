// react library
import React from 'react'

// styles
import './Content.scss'

// components
import BorderBox from '../BorderBox/index'

// mock data
import file from '../../mockData/employees.js'

export default class Content extends React.Component {
  render() {
    let department = file.data;

    const colors = ['#bd10e0', '#4a90e2', '#50e3c2', '#b8e986', '#7ed321', '#417505', '#f8e71c', '#f5a623', '#9b9b9b']
    const totalEmployees = Object.values(department).reduce((acc, data) => acc + data.length, 0)
    console.log(totalEmployees, '=====')

    return (
      <div className="content-container">
        <div className="content-container__content">
          <h1 className="content-container__content__heading">Departments</h1>
          <div className="content-container__content__items">
            {
              Object.entries(department).map(([key, value], index) => {
                // initially boxwidth
                // so both height and width was gotten from here 
                // then a min-width was set on the container so the boxes don't grow more than that
                const boxHeight = `${(value.length / totalEmployees) * 100}%`
                const boxWidth = `${value.length / totalEmployees * 100}%`
                // use the dom height to get padding for each box
                const padding = `${51 * (index)}px`;
                // const padding = domHeight;
                return <BorderBox
                  // boxWidth={boxWidth}
                  padding={padding}
                  colors={colors[index % colors.length]}
                  boxHeight={boxHeight}
                  key={index}
                  data={value}
                  name={key}
                  boxWidth={boxWidth}
                // department={department}
                />
              })}
          </div>
        </div>
      </div>
    )
  }
}
