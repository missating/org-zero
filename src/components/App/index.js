// react library
import React from 'react';

// styles
import './App.css';

// components
import NavBar from '../NavBar'
import Content from '../Content'

const App = () => {
  return (
    <div className="app-container">
      <NavBar />
      <Content />
    </div>
  )
}

export default App;
