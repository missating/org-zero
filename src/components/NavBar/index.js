// react library
import React from 'react';

// images
import ZeroLogo from '../../assets/Zero.png';

// styles
import './NavBar.scss'

const NavBar = () => {
  return (
    <header className="navbar-container">
      <div className="navbar-container__content">
        <div className="navbar-container__content__logo">
          <img src={ZeroLogo} alt="org-zero-logo" width="100" />
        </div>
        <nav>
          <ul className="navbar-container__content__items">
            <li>About</li>
            <li>FAQs</li>
          </ul>
        </nav>
      </div>
    </header>
  )
}

export default NavBar;
