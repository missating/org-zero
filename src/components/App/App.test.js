// react library
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';

// components
import App from './index'
import NavBar from '../NavBar'
import Content from '../Content'

describe('App Component', () => {
  it('should be rendered properly', () => {

    const wrapper = shallow(<App />);
    expect(wrapper.find('div').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should mount both the Navbar and Content Components when on /', () => {
    const location = {
      pathname: '/',
    };

    const wrapper = shallow(<App location={location} />);

    expect(wrapper.find(NavBar).length).toBe(1);
    expect(wrapper.find(Content).length).toBe(1);
  });
})

