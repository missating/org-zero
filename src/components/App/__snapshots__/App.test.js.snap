// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should be rendered properly 1`] = `ShallowWrapper {}`;
